### DISCUSSION TITLE
Osu selects song different to one that's playing when entering song selection

### DISCUSSION THREAD

**Discussion by Cootz** [ppy/osu#22692] [Q&A]
Is it a bug or a feature?

https://user-images.githubusercontent.com/50776304/219972958-44a6090d-b3f6-4d90-bbd7-20ce34e92491.mp4

It quite annoys when you want to play song you found in player but you can't just hop to it. You have to memorize name and find map by your own

**Comment by mk56-spn** [comment]
Iirc this happens when the song in the main menu isn't available in the ruleset you have selected / you don't have converts on.

**Reply by Axilotl17** [reply]
Hmm, it happened for me as shown in the video, where the map playing was in std when I was too but now I can't recreate it.

**Reply by monochrome22** [reply]
Did you have star rating filter on?

**Comment by monochrome22** [comment]
Same thing #22500

**Reply by Cootz** [reply]
Not sure about it. Paradigm Shift have difficulties that correspond gamemode and filter.
![image](https://user-images.githubusercontent.com/50776304/220102665-feeb9616-bd29-48db-944f-ac7d319b41f5.png)

**Reply by monochrome22** [reply]
The song player selects a certain difficulty, from what I can tell, it's in alphabetical order. So if the difficulty selected is out of star rating range, then it will change to a random map once you enter song select. 
For Paradigm Shift, it selects the "Advanced" difficulty, which is 2.66 stars. 

In case you are thinking of suggesting to make it select a viable difficulty: that would only half solve the problem, because the player would still play songs for other game modes and those that don't have any viable difficulties, so the song would still change when entering song select. 

The proper solution would be to only have the player play songs that match the current game mode and filters and select a viable difficulty, also smoothly transition to another song if necessary when changing game mode in main menu.

**Reply by Cootz** [reply]
> the player would still play songs for other game modes and those that don't have any viable difficulties

Tbh I like this feature. It just looks stupid when you try to play song from player, enter beatmap selection, song disappears but you type in search `BeatmapSet` name and find it :/

**Reply by Cootz** [reply]
Song player breaks filter for `Beatmap` completely lol

https://user-images.githubusercontent.com/50776304/220112820-f9b0c26b-17c4-4b6d-9538-7cefd8b7ac70.mp4
